packages/backend/src/server/routers/campaign.ts

```diff
@@ -33,42 +32,41 @@ export const campaignRouter = router({
       } else if (!isKyc(ctx.session.user)) {
         throw new Error('MissingKYC')
       }
-      if (input.address) {
-        const campaign = await prisma.campaign.findFirst({
-          where: {
-            address: {
-              equals: input.address.toLowerCase(),
-              mode: 'insensitive',
-            },
-            chainId: input.chainId,
+
+      const campaign = await prisma.campaign.findFirst({
+        where: {
+          address: {
+            equals: input.address.toLowerCase(),
+            mode: 'insensitive',
           },
-        })
+          chainId: input.chainId,
+        },
+      })
 
-        if (campaign) {
-          throw new Error('CampaignExists')
-        }
-      } else if (input.chainId && input.user) {
-        const chain = getChain(input.chainId)
+      if (campaign) {
+        throw new Error('CampaignExists')
+      }
 
-        if (!chain) {
-          throw new Error('MissingChainId')
-        }
+      const chain = getChain(input.chainId)
 
-        const provider = new providers.JsonRpcProvider(chain.rpcUrls.default)
+      if (!chain) {
+        throw new Error('MissingChainId')
+      }
 
-        if (!chain.campaignFactory) {
-          throw new Error('MissingCampaignFactory')
-        }
+      if (chain.campaignFactory) {
+        const provider = new providers.JsonRpcProvider(chain.rpcUrls.default)
 
-        const factory = new Contract(chain.campaignFactory, campaignFactoryAbi, provider)
+        const contract = new Contract(chain.campaignFactory, campaignFactoryAbi, provider)
 
-        const userCampaigns = await factory.getUserCampaigns(input.user)
+        const attached = contract.attach(input.address)
```

Comment:
From here on things go sideways. You already had the code, simply make sure getUserCampaigns return the campaign specified in the request.

You are overwriting the campaign factory with the campaign address and then query who deployed it. This still means the user could have deployed any random contract and it would be set as campaign. Only campaigns deployed by our campaign factory are allowed.

Reply:
not sure that this works based on returning BigNumber values in an array, see the following commit and let me know your thoughts.